<a href="https://colab.research.google.com/github/ANGIECATHERINEP/MujerDigital/blob/main/Clase2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**OPERADORES ARITMÉTICOS**

---
Los operadores permiten realizar diferentes procesos de calculo en cualquier lenguaje de programación.

Los operadores más basicos son: 
1. suma
2. Resta
3. Multiplicación 
4. División





**Suma y resta**

---
simbolo suma(+) el cual se utilizara en medio de la declaracion de las varianles a operar.

Simbolo resta(-)


In [3]:
print(10+300)
print (20-50)

310
-30


In [4]:
a = 20
b = 35
print(a+b)
print(a-b)

55
-15


In [5]:
numero_botella = 12
print(numero_botella + 8)

20


In [6]:
numero_botella = 12
numero_botella = numero_botella + 8
print(numero_botella)

20


**Sumas y restas entre dos diferentes tipos de variables**

---
los tipos de variables:
1. Declaración de variables númericas enteros
2. Flotantes(incorpora y flexibiliza lols decimales)
3. Tipo cadena


In [7]:
q1 = 100
q2 = 24.5
print(type(q1+q2))

<class 'float'>


In [12]:
v1 = "quiero"
v2 = "un celular"
v3 =v1 +" " + v2
print(v1+v2)
print(v1,v2)
print(v3)

quieroun celular
quiero un celular
quiero un celular


In [14]:
xq2 = 235
xq3 = 400
print(xq2-xq3)

xq2 = 24.66
xq3 = 4
print(type(xq2-xq3))

-165
<class 'float'>


**IMPORTANTE**

La resta no funciona directamente para variables tipo cadena.

**Multiplicación**

---
Definamos la multiplicación como la suma de varias veces un numero, cantidad de veces indicadas por otro numero (4x2=4+4=8). El simbolo es *.


In [16]:
cali = 5
palmira = 6
print(cali*palmira)

30


**IMPORTANTE**

Se aplica la ley conmutativa.
Recordemos que la ley conmutativa es que el orden de los factores no altera el resultado (aplica en suma y multiplicación)

In [17]:
cali = 5
palmira = 6
print(palmira*cali)

30


**División**

---
ACLARACIÓN: **NO** aplica la ley conmutativa. 

el simbolo es /.


In [19]:
a = 20
b = 4
print(a/b, b/a)

5.0 0.2


Para el print podemos separa las operaciones por variable con (,), lo cual no afectara la impresión de los resultados.

**Tipos de división**

---
1. División exacta con(//)
2. División inexacta o con decimales con (/)




In [22]:
print(15//5)
print(15/4)

3
3.75


Nota: La opración de división, al igual que una división normal no es posible dividir entre cero (0); Porque en matematicas es un resultado indefinido.

4. Potenciación
5. Módulo

**Potenciación**

---
En Python se maneja el simbolo (**). El primer valor es la base y el ultimo es el exponente


In [24]:
print(2**3)

8


**Módulo**

---
Le llamamos módulo al residuo de la opración de división. El simbolo usado para mostrar el residuo es (%); El primer valor es el dividendo y el segundo el divisor


In [26]:
print(88%9)

7


**OPERADORES DE COMPARACIÓN**

---
Los opradores de comparación sirven para comparar varios valores.
En la vida cotodiana a la hora de tomar decisiones teemos que decidir entre un camino u otro, ejemplo: en las compras, desiciones de vida, caminos por recorrer.




**Igualdad**

---
Simbolo en Python es (==)


In [27]:
x = 30
z = 31
print(x==z)

False


In [28]:
x = "casa"
z = "casa"
print(x==z)

True


**Diferencia**

---
A diferencia del operador de igualda el sismbolo esta combinado por el simbolo de admiración, es decir (!=).


In [29]:
x = "casa"
z = "caza"
print(x!=z)

True


**Mayor que**

---
Símbolo en Python (>).
Siempre comprativa se designa cual es el menor o el mayor , por el valor que esta 
a la izquierda de la opración.


In [30]:
print(30>29.8)

True


**Menor que**

---
Caso contrario, el simbolo es (<)


In [31]:
x = 4
y = 3
print(x<y)

False


**Mayor o igual que**


---
Es similar al mayor que, con la diferencia de que la forma en dara el resultado verdadero es que si ambos números son iguales y flexibiliza dando verdadero de igual forma si el valor de la izquierda es mayor que el de la derecha.

El simbolo es (>=)


In [32]:
x = 10
y = 10
print(x>=y)

True


In [33]:
x = 10
y = 5
print(x>=y)

True


**Menor o igual que**

---

el símbolo es (<=)

In [34]:
c = 45
print(c<=46)

True


**tener en cuenta:** En la comparaciones los resultados que nos arrojan siempre es de tipo booleano es decir: True o false 

**OPERADORES LÓGICOS**

---
Python incluye tres operadoress lógicos base:
1. AND(y)
2. OR(o)
3. NOT(no)

los operadores lógicos funcionan con valores booleanos a su vez, devuelven como respuesta valores booleanos


**Operadores AND**

---
Este operador solo devolverá (verdadero) si ambos valores lo son.

¿como usar este operador?
Basta con escribir un valor booleano, luego la palabra "and" y por ultimo otro valor booleano


In [35]:
print(True and False)

False


**Ejemplo de contextualización AND**

1. True and True = true
2. True and False = False
3. False and False = False
4. False and True False = False

**Operador OR**

---
A difrencia del operador AND, or devleve a todo verdadero a menos de que ambos valores sean falsos.


In [37]:
print(True or True)
print(False or False)

True
False


Ejemplo  de contextualización OR

1. True or True = true
2. True or False = true
3. False or True  = true
**4. False or False = False**

In [39]:
a = 3
b = 2
print(a>4 or a==4)
print(b==b and a>=b)

False
True


**Operador NOT**

---
Este operador es negación, en español NO.

Este operador es muy especial, dado que por si solo sirve para hacer ninguna operación, lo único que hace es negar, o **invertir el valor** que viene despues de él.


In [40]:
print(not False)
print(not True)

True
False


**PRECEDENCIA DE OPERADORES**

---
Es muy importante manejar el orden y las opraciones al realizarlas, para no caer en problemas código error.

El orden debe ser:
1. De izquierda a derecha
2. Primero los parentesis
3. Potenciación o radicación
4. Multiplicacion y división
5. Suma y resta

**ejemplo**

4 - 2 (10) = -16
1. Primero el parentesis: 4 - 2 * 10=
2. Se continúa con la multiplicación: 4 - 20 =
3. Por último la resta: 4-20= -16

*¿Cual crees que es el resultado de la siguiente operación?*

In [41]:
a = False
b = True
c = True
print(not c and (a or b))

False


In [43]:
print(3-2*(13-5))

-13


**RESUMEN**

---
1. Aplicación de ley conmutativa: Solo para suma y resta (operadores básicos)
2. los resultados de los operadores lógicos son de tipo booleano.
3. NO se puede realizar resta entre variables del tipo cadena (el procedimiento si abarca numeros es la conversion del inicio)
4. Si aplicamos el condicional AND, para el que resultado sea TRUE, la única forma es que ambos valores sean verdaderos.
5. En cambio con el OR, si cumple que alguno de los dos valores son TRUE, el resultado será TRUE.




-



---



---


#**TAREA CLASE N° 2**

---


---





**a.**

In [47]:
print((8 - 2) * 15 // 3)

30


1. Primero el parentesis: ( 6 )* 15 / 3 =
2. Se continúa con la multiplicación: 6 * 15 / 3 =
3. Por último la división: 90 / 3= 30

**b.**

In [53]:
a = 35
b = 50
c = 28
print(a>=c and (b>=c and a!= b))

True


1. Primero los valores del parentesis: 

    b >= c --- True

    a != b --- True

    True **AND** True = True

2. Luego se resuelve: 

    a >= c ---True

3. Por último: 

    True AND True = True

**c.**

In [49]:
print(4**3*(10 + 5))

960


1. Primero el parentesis: 4**3 *(15)=
2. Se continúa con la potenciación : 64*(15)=
3. Por último con la multiplicación: 64 * 15= 960


**d.**

In [50]:
x = 5
y = 4.5
z = 6
print(x==y or (z>x and y>z))

False


1. Primero los valores del parentesis: 

    z > x = True     **AND**      y > z = false

    True  **AND**   False = False

2. Se continúa con los valores del lado izquierdo: 

    x == y ---False

3. Por último: 

    False **OR**  False = False

**e.**

In [54]:
u = False
v = True
w = False
print(not w or (u and v))

True


1. Primero el parentesis:
   
    u **AND** v = False
    
2. Por último:  

    **NOT** w = True

    False **OR** True = True